File: Day_03/solveA.jl
x = [line for line in readlines(open("example.txt"))]

function processID(line)
	return(split(replace(replace(strip(split(line, '@')[2]), ": " => ","), "x" => ","), ','))
end

Xgaps = []
Ygaps = []
Xwidths = []
Ywidths = []

for line in readlines(open("example.txt"))
	Xgap, Ygap, Xwidth, Ywidth = processID(line)
	append!(Xgaps, parse(Int, Xgap))
	append!(Ygaps, parse(Int, Ygap))
	append!(Xwidths, parse(Int, Xwidth))
	append!(Ywidths, parse(Int, Ywidth))
end

maxX = 1000
maxY = 1000

fabric = [[0 for y in range(1, length=maxX)] for x in range(1, length=maxY)]

for i in range(1, length=length(Xgaps))
	for x in range(Xgaps[i], length=Xgaps[i]+Xwidths[i])
		for y in range(Ygaps[i], length=Ygaps[i]+Ywidths[i])
			fabric[x][y] += 1
		end
	end
end

print(sum([sum([square >= 2 for square in line]) for line in fabric]), '\n')

File: Day_03/solveA.py
import numpy as np

def processID(line):
	return(line.split('@')[1].strip().replace(': ',',').replace('x',','))

Xgaps = []
Ygaps = []
Xwidths = []
Ywidths = []

for line in open('input.txt'):
	Xgap, Ygap, Xwidth, Ywidth = processID(line).split(',')
	Xgaps.append(int(Xgap))
	Ygaps.append(int(Ygap))
	Xwidths.append(int(Xwidth))
	Ywidths.append(int(Ywidth))

assert len(Xgaps) == len(Ygaps) == len(Xwidths) == len(Ywidths), 'Length mismatch'

maxX = max(np.array(Xgaps) + np.array(Xwidths))
maxY = max(np.array(Ygaps) + np.array(Ywidths))
fabric = np.zeros((maxX, maxY))

for i in range(len(Xgaps)):
	for x in range(Xgaps[i], Xgaps[i]+Xwidths[i]):
		for y in range(Ygaps[i], Ygaps[i]+Ywidths[i]):
			fabric[x][y] += 1

#print(fabric)
print(sum([sum([square >= 2 for square in line]) for line in fabric]))
